tensorboard/webapp/app_routing/effects/app_routing_effects_test.ts

```diff
@@ -334,6 +334,45 @@ describe('app_routing_effects', () => {
         );
       });
 
+      it('does not warn user when changing tab (same routeId)', fakeAsync(() => {
+        spyOn(window, 'confirm');
+        getPathSpy.and.returnValue('/experiments');
+        // Changing tab.
+        getHashSpy.and.returnValue('#foo');
+        action.next(
+          actions.navigationRequested({
+            pathname: '/experiments',
+          })
+        );
+        tick();
```

Comment:
Looking at this spec alone, it is unclear whether this is supposed to have the same routeId as current one. Can you overrideSelector or setup so that is is more explicit here?

Reply:
Done.